models/conditional_gans.py

```diff
@@ -65,13 +68,98 @@ class StackGAN1(ConditionalGAN):
     """ Definition for the stage 1 StackGAN """
     def __init__(self, img_size, num_latent_dims, kernel_size,
                  num_filters, reshape_dims):
-        super().__init__(img_size, num_latent_dims, kernel_size,
-                         num_filters, reshape_dims)
+
+        generator = GeneratorStage1(
+            img_size=img_size,
+            num_latent_dims=num_latent_dims,
+            kernel_size=kernel_size,
+            num_filters=num_filters,
+            reshape_dims=reshape_dims
+        )
+
+        discriminator = DiscriminatorStage1(
+            img_size=img_size,
+            kernel_size=kernel_size,
+            num_filters=num_filters
+        )
```

Comment:
Why are these before calling `super`?

Reply:
they need to be passed into super because each stage has different generators and discriminators. Tbh I do not know if we really need the parent class here, but I just left it for now.

Reply:
OK, sounds good :+1: